demo/src/lt.tooltip.js

```diff
@@ -73,19 +116,31 @@ angular.module('lt.tooltip', [])
                 $tooltipEl = angular.element(el.querySelector('.tooltip')),
                 direction = $attrs.direction || TOOLTIP_DEFAULT_DIRECTION,
                 template = $attrs.template || $attrs.content || TOOLTIP_DEFAULT_TEXT,
+                directions = DEFAULT_DIRECTION_PRIORITY,
                 dimensions,
-                position;
+                position,
+                withinVisibleArea,
+                counter = 0;
 
             // Make sure the template has been put in the DOM before retrieval
             // of dimensions. Otherwise, the dimensions will be reported
             // (incorrectly) without the content.
             $tooltipEl.html(template);
-
-            dimensions = getDimensions($el),
-            position = calculatePosition(dimensions, direction);
-
-            $tooltipEl.css('top', position.top);
-            $tooltipEl.css('left', position.left);
+            dimensions = getDimensions($el);
+
+            // Recalculate until tooltip is within visible area.
+            while (!withinVisibleArea && counter < directions.length) {
+                position = calculatePosition(dimensions, direction);
+                withinVisibleArea = isWithinVisibleArea(position, dimensions);
+                if (!withinVisibleArea) {
+                    directions = removeDirection(direction, directions);
+                    direction = nextDirection(direction, directions);
+                }
```

Comment:
You don't need counter. Since you're calling `removeDirection` which removes an element from the `directions` array, simply checking to see if that array has a length will suffice.